rclpy/rclpy/node.py

```diff
@@ -1194,6 +1197,8 @@ def add_on_set_parameters_callback(
         :param callback: The function that is called whenever parameters are being validated
                          for the node.
         """
+        if not callable(callback):
+            raise TypeError('Callback must be callable, got {}', type(callback))
```

Comment:
we can do type hint checking here something like,

```diff
+        if inspect.signature(callback).return_annotation is not SetParametersResult:
+            warnings.warn('Callback does not have the correct return annotation. '
+                          'Callback must return a SetParametersResult.')
```

but this breaks the downstream code if those do not have annotation for return type. (and also breaks the existed tests in `rclpy`.)

so i would suggest we can check the actual return value and then print the warning for the user application, instead of enforcing the type annotation.